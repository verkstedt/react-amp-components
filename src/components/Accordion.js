import React, { Component } from 'react'
import { oneOfType, arrayOf, node, bool } from 'prop-types'
import { Helmet } from 'react-helmet'

import pick from '../utils/pick'
import commonPropTypes from './commonPropTypes'

class Sidebar extends Component {
  static propTypes = {
    ...commonPropTypes,
    disableSessionStates: bool,
    expanded: bool,
    children: oneOfType([
      arrayOf(node),
      node
    ])
  }

  static defaultProps = {
    disableSessionStates: null,
    expanded: null,
    children: []
  }

  get properties() {
    return pick(this.props, ...[
      ...Object.keys(commonPropTypes),
      'disableSessionStates',
      'expanded'
    ])
  }

  render() {
    const { children } = this.props
    return [
      <Helmet key="helmet">
        <script async="" custom-element="amp-accordion" src="https://cdn.ampproject.org/v0/amp-accordion-0.1.js" />
      </Helmet>,
      <amp-accordion {...{ ...this.properties }} key="accordion">
        {children}
      </amp-accordion>
    ]
  }
}

export default Sidebar

