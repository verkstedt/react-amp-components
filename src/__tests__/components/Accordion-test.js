import React from 'react'
import Accordion from '../../components/Accordion'
import Helmet from '../../utils/Helmet'
import { renderComponent } from '../test-utils'

describe('Accordion', () => {
  it('matches the snapshot', () => {
    const accordion = renderComponent(
      <Accordion>
        <section>
          <h2>Section 1</h2>
          <p>Content in section 1.</p>
        </section>
        <section expanded="">
          <h2>Section 2</h2>
          <div>Content in section 2.</div>
        </section>
      </Accordion>
    )
    expect(accordion.toJSON()).toMatchSnapshot()
  })

  it('injects the right script tag', () => {
    renderComponent(
      <Accordion>
        <section>
          <h2>Section 1</h2>
          <p>Content in section 1.</p>
        </section>
        <section expanded="">
          <h2>Section 2</h2>
          <div>Content in section 2.</div>
        </section>
      </Accordion>
    )
    Helmet.canUseDOM = false
    const staticHead = Helmet.renderStatic()
    expect(staticHead.scriptTags).toMatchSnapshot()
  })

  it('passes down the right props', () => {
    const props = {
      expanded: true,
      disableSessionStates: true
    }
    const accordion = renderComponent(
      <Accordion {...props}>
        <section>
          <h2>Section 1</h2>
          <p>Content in section 1.</p>
        </section>
        <section expanded="">
          <h2>Section 2</h2>
          <div>Content in section 2.</div>
        </section>
      </Accordion>
    )
    expect(accordion.toJSON().props).toMatchObject(props)
  })
})

