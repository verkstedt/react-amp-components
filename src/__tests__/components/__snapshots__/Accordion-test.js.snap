// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion injects the right script tag 1`] = `
Array [
  Object {
    "async": "",
    "custom-element": "amp-accordion",
    "src": "https://cdn.ampproject.org/v0/amp-accordion-0.1.js",
  },
]
`;

exports[`Accordion matches the snapshot 1`] = `
<amp-accordion
  disableSessionStates={null}
  expanded={null}
>
  <section>
    <h2>
      Section 1
    </h2>
    <p>
      Content in section 1.
    </p>
  </section>
  <section
    expanded=""
  >
    <h2>
      Section 2
    </h2>
    <div>
      Content in section 2.
    </div>
  </section>
</amp-accordion>
`;
